src/components/SideSheet.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+	 <div
+		v-if="value"
+	 	id="overlay"
+		tabindex="0"
+		@click="shouldCloseOnBackdrop"
+	>
+		<div
+			id="container"
+			:class="floatClass"
+			@click.stop
+		>
+			<slot name="close-icon">
+				<div
+					class="text-right pb-2"
+				>
+					<x-icon
+						id="close-icon"
+						@click.stop="$emit('input', !value)"
+					/>
+				</div>
+			</slot>
+			<slot/>
+		</div>
+	</div>
+</template>
+
+<script>
+import { XIcon } from 'vue-feather-icons';
+export default {
+	components: {
+		XIcon,
+	},
+	props: {
+		value: {
+			type: Boolean,
+			default: false,
+			description: `
+				Prop to control if the SideSheet should be shown or not (usually by v-model)
+			`,
+		},
+		right: {
+			type: Boolean,
+			default: false,
+			description: `
+				Prop to define that the element will be shown at right screen side.
+			`,
+		},
+		left: {
+			type: Boolean,
+			default: false,
+			description: `
+				Prop to define that the element will be shown at left screen side (right ifs default).
+			`,
+		},
+		noCloseOnBackdrop: {
+			type: Boolean,
+			default: false,
+			description: `
+				Prop to define if the element won't be dismissed when backdrop is click.
+			`,
+		},
+		noCloseOnEsc: {
+			type: Boolean,
+			default: false,
+			description: `
+				Prop to define if the element won't be dismissed when esc pressed.
+			`,
+		}
+	},
+
+	computed: {
+		floatClass() {
+			if (this.left) {
+				return 'left';
+			}
+			if (this.right) {
+				return 'right';
+			}
+			return 'right';
+		}
+	},
+
+	mounted() {
+		let self = this; 
+
+		window.addEventListener('keyup', function(ev) {
+				if (ev.keyCode === 27) { // esc
```

Comment:
I think that it would be better for readability to use a constant value for the key code, like

```suggestion
				if (ev.keyCode === keyCodes.ESC) {
```

or

```suggestion
				if (ev.keyCode === ESC) { // esc
```

What do you think?